opm/simulators/linalg/bda/Matrix.hpp

```diff
@@ -17,41 +17,83 @@
   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#ifndef FPGA_MATRIX_HEADER_INCLUDED
-#define FPGA_MATRIX_HEADER_INCLUDED
+#ifndef MATRIX_HEADER_INCLUDED
+#define MATRIX_HEADER_INCLUDED
 
 #include <vector>
 
+#include <opm/simulators/linalg/bda/opencl.hpp>
+#include <opm/simulators/linalg/bda/BlockedMatrix.hpp>
+
 namespace Opm
 {
 namespace Accelerator
 {
 
+class Matrix;
+
+/// This struct resembles a csr matrix, only doubles are supported
+/// The matrix data is stored in OpenCL Buffers
+class OpenclMatrix {
```

Comment:
Please change the filenames to reflect the class name

Reply:
I'll create new files for `OpenclMatrix`.

Reply:
3b8bb461